internal/appsec/emitter/httpsec/http.go

```diff
@@ -79,9 +79,12 @@ func WrapHandler(handler http.Handler, span ddtrace.Span, pathParams map[string]
 		var blocking bool
 		args := MakeHandlerOperationArgs(r, clientIP, pathParams)
 		ctx, op := StartOperation(r.Context(), args, func(op *types.Operation) {
-			dyngo.OnData(op, func(a *sharedsec.Action) {
-				bypassHandler = a.HTTP()
-				blocking = a.Blocking()
+			dyngo.OnData(op, func(a *sharedsec.HTTPAction) {
+				blocking = true
+				bypassHandler = a.Handler
+			})
+			dyngo.OnData(op, func(a *sharedsec.StackTraceAction) {
+				// TODO: do something with the stacktrace
```

Comment:
Even though it would work to collect the stack trace over here, I don't think this is the right "logical" place to handle this action. I'd put it in a "standalone/generic dygno listener" that we can plug to any dyngo event listener (like a "middleware").

```suggestion
```

Reply:
Just to make sure: the stacktrace is not collected there. This is where the generated stacktrace data would be retrieved to be added to the span